presto-matching/src/test/java/com/facebook/presto/matching/TestMatcher.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.matching;
+
+import com.facebook.presto.example.ast.FilterNode;
+import com.facebook.presto.example.ast.JoinNode;
+import com.facebook.presto.example.ast.ProjectNode;
+import com.facebook.presto.example.ast.RelNode;
+import com.facebook.presto.example.ast.ScanNode;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static com.facebook.presto.example.ast.Patterns.filter;
+import static com.facebook.presto.example.ast.Patterns.plan;
+import static com.facebook.presto.example.ast.Patterns.project;
+import static com.facebook.presto.example.ast.Patterns.scan;
+import static com.facebook.presto.example.ast.Patterns.source;
+import static com.facebook.presto.example.ast.Patterns.tableName;
+import static com.facebook.presto.matching.Capture.newCapture;
+import static com.facebook.presto.matching.Pattern.any;
+import static com.facebook.presto.matching.Pattern.typeOf;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.expectThrows;
+
+public class TestMatcher
+{
+    @Test
+    void trivialMatchers()
```

Comment:
Would rather not as it doesn't matter - those are only called by the test framework. I even like it more without `public` here as it only makes it slower for the eyes to get to the test name :). I'd prefer to avoid reediting for such trivia.